scripts/lib/robot.sh

```diff
@@ -33,15 +33,29 @@ execute_robot() {
   # _test_path can be either
   #   - path to directory containing a set of .robot files
   #   - path to a single .robot file
-  local _test_path=${*}
-  _test_path=${_test_path%%--*}
-  local _robot_args=${*}
-  _robot_args=${_robot_args#*--}
-  if ! echo "${*}" | grep " -- "; then
-    _robot_args=""
-  fi
-  local _test_name
-  _test_name="$(echo "$_test_path" | sed -e 's/\ /_/' | awk '{$1=$1};$1')"
+  local _args=("$@")
+  local _test_path=()
+  local _seperator_idx=-1
+  local _args_len=${#_args[@]}
+  _test_path=()
+
```

Comment:
@philipandag Would it make sense to add some comments with these blocks, to explain what your goal is?

Reply:
Yes, it most definitely would make sense. I will explain it a bit.

Reply:
https://github.com/Dasharo/open-source-firmware-validation/pull/465/commits/6b75d6c1af325478c85c72c5386c347e346aef8a